include/rpm/rpmfileutil.h

```diff
@@ -107,6 +107,8 @@ char * rpmGenPath	(const char * urlroot,
  */
 char * rpmGetPath (const char * path, ...) RPM_GNUC_NULL_TERMINATED;
 
+#define rpmCatPath(...)  rpmCleanPath(rstrscat(NULL, ##__VA_ARGS__))
```

Comment:
`###__VA_ARGS` is a gcc extension, we can't use such a thing unconditionally (but if this is just a temporary shortcut before refactoring then nevermind)

Reply:
Oh, I didn't know that, so thanks for pointing it out. It was meant for "real" :smile: Do you know if there's a way to accomplish the same without a compiler extension? I was contemplating an actual function, but that turned out to be unnecessarily complicated so went with a macro instead.

Reply:
Also it's worth mentioning here that it's meant purely as syntactic sugar, there's no reason why we couldn't just call both functions nested like this, in those (so far) two places where it happens.

Reply:
No idea, I'm pretty clueless when it comes to preprocessor tricks.
I'd just place the nested calls in the two places that need it for now, and later add a proper API that can handle paths with or without canonicalization, expansion etc.

Reply:
Yeah, I toyed with the idea of having one function do all of that and control it with flags, pretty much like `rpmGetPath` but with a bunch of `if`s, but at that point, it's basically the same as chaining those individual functions (for canonicalization, expansion etc.) together in the caller, which also has the benefit of using whatever functions, not just those supported by the hypothetical flags parameter. But sure, there could be a way to make this into a nicer API eventually.

For now, let's just go with the simplest way, as you suggest.

Reply:
This being Friday... just for giggles:
https://github.com/pmatilai/rpm/commit/9d3050fe192a8a5abb1c40e19deebd4a69c11ff2

Reply:
Code is worth a thousand more than empty words, that looks good, thanks! :smile: That'll make for a nice addition to this PR (whichever lands first).